app/controllers/setup_controller.rb

```diff
@@ -29,17 +31,22 @@ def welcome
     @services_cidr = Pillar.value(pillar: :services_cidr) || "172.24.0.0/16"
     @api_cluster_ip = Pillar.value(pillar: :api_cluster_ip) || "172.24.0.1"
     @dns_cluster_ip = Pillar.value(pillar: :dns_cluster_ip) || "172.24.0.2"
+    @suse_registry_mirror = DockerRegistry.first || DockerRegistry.new
```

Comment:
Are we sure `DockerRegistry.first` is always going to return the SUSE registry mirror? Wouldn't be safer to use something like `find_or_initialize_by`?

Reply:
At this point I don't think that we would have any registry stored in the database. So it would kind of be the same if I use by `find_or_initialize_by(mirror: SUSE_MIRROR_REGISTRY)`.

Reply:
I was thinking about a user that goes back to the setup page after the cluster has been deployed and some registries have been added/removed. I would prefer to play safe and change the code to use `find_or_initialize_by`

Reply:
User can't go back to the setup pages after bootstrap unless they hack our code.

Reply:
I see. If you want to do the change go ahead, otherwise leave the code as it is.